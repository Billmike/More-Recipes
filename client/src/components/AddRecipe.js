import React from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

const AddRecipe = (props) => {
    console.log(props)
	return (
	   <div>
       <h1 className="container">Add Recipe</h1>
        <RecipesForm onSubmit={(recipe) => {
            props.dispatch(addRecipe(recipe));
            props.history.push('/')
        }} />
       </div>	
	)
}

export default connect()(AddRecipe);
