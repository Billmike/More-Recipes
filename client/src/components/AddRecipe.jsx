import React, { Component } from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

export class AddRecipe extends Component {
  onSubmit = (recipe) => {
    this.props.addRecipe(recipe);
    this.props.history.push('/dashboard');
  };

  render() {
    return (
      <div>
        <h1 className="container"> Add Recipe </h1>
        <RecipesForm
          onSubmit={this.onSubmit}
        />
      </div>
      );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addRecipe: (recipe) => dispatch(addRecipe(recipe))
  }
}

// const AddRecipe = (props) => {
//     console.log(props)
// 	return (
// 	   <div>
//        <h1 className="container">Add Recipe</h1>
//         <RecipesForm onSubmit={(recipe) => {
//             props.dispatch(addRecipe(recipe));
//             props.history.push('/')
//         }} />
//        </div>	
// 	)
// }

export default connect(undefined, mapDispatchToProps)(AddRecipe);
