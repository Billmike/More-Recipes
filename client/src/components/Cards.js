import React, { Component } from 'react';

class Cards extends Component {
	render () {
		return (
			<div>
			  <div className="card-group">
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">180 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="border border-secondary rounded btn" data-toggle="modal" data-target="#exampleModal">  150 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 70 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action. </p>
              <a href="./template/recipe_detail.html" className="border btn border-secondary rounded">67 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="border btn border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  49 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 90 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
        </div>
		</div>
		)
	}
}

export default Cards;
