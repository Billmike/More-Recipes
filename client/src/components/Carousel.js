import React, { Component } from 'react';

class Carousel extends Component {
	render () {
		return (
			<div>
			  <div id="carouselExampleIndicators" className="carousel slide" data-ride="carousel">
          <ol className="carousel-indicators">
            <li data-target="#carouselExampleIndicators" data-slide-to="0" className="active"></li>
            <li data-target="#carouselExampleIndicators" data-slide-to="1"></li>
            <li data-target="#carouselExampleIndicators" data-slide-to="2"></li>
          </ol>
          <div className="carousel-inner">
            <div className="carousel-item active">
              <img className="d-block w-100 sizing" src="../src/assets/img/pancakes.jpeg" alt="First slide" />
              <div className="carousel-caption d-none d-md-block">
                  <h3 className="carousel-h3">Handcrafted recipes made by local chefs</h3>
                  <p className="carousel-p">... the little things matter</p>
                </div>
            </div>
            <div className="carousel-item">
              <img className="d-block w-100 sizing" src="../src/assets/img/cookies.jpeg" alt="Second slide" />
              <div className="carousel-caption d-none d-md-block">
                  <h3 className="carousel-h3-light">Tasty recipes to spice up your dinner table.</h3>
                  <p className="carousel-p-light">... you really can't resist these delights.</p>
                </div>
            </div>
            <div className="carousel-item">
              <img className="d-block w-100 sizing" src="../src/assets/img/dessert.jpeg" alt="Third slide" />
              <div className="carousel-caption d-none d-md-block">
                  <h3 className="carousel-h3">Make every meal memorable.</h3>
                  <p className="carousel-p">... no more boring nights at home.</p>
                </div>
            </div>
          </div>
          <a className="carousel-control-prev" href="#carouselExampleIndicators" role="button" data-slide="prev">
            <span className="carousel-control-prev-icon" aria-hidden="true"></span>
            <span className="sr-only">Previous</span>
          </a>
          <a className="carousel-control-next" href="#carouselExampleIndicators" role="button" data-slide="next">
            <span className="carousel-control-next-icon" aria-hidden="true"></span>
            <span className="sr-only">Next</span>
          </a>
        </div>
			</div>
		)
	}
}

export default Carousel;
