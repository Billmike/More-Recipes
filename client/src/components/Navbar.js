import React, { Component } from 'react';
import { connect } from 'react-redux';

class Navbar extends Component {
  render() {
    return (
      <div>
        <nav className="navbar navbar-expand-lg navbar-light bg-light navbar-fixed-top">
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#startupNavbar" aria-controls="navbarTogglerDemo01" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <a className="navbar-brand" href="#"><img src="img/logo.png" alt="More-Recipes" height="26" /></a>
        <div className="collapse navbar-collapse navbar-expand" id="startupNavbar">
          <ul className="nav navbar-nav ml-auto">
              <li className="nav-item active">
              <a className="nav-link" href="#">Home <span className="sr-only">(current)</span></a>
            </li>
              <li className="nav-item">
              <a className="nav-link" href="" data-toggle="modal" data-target="#sigininmodal">Sign In</a>
            </li>
              <li className="nav-item">
              <a className="btn btn-primary" href="" data-toggle="modal" data-target="#signupmodal">Sign Up</a>
            </li>
            </ul>
        </div>
            </nav>
        <div className="modal fade" id="sigininmodal" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
  <div className="modal-dialog" role="document">
    <div className="modal-content">
      <div className="modal-header">
        <h5 className="modal-title" id="exampleModalLabel">Sign in</h5>
        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div className="modal-body">
        <form>
                          <div className="form-group">
                              <input type="email" className="form-control form-control-lg" placeholder="Email Address" />
                            </div>
                          <div className="form-group">
                              <input type="password" className="form-control form-control-lg" placeholder="Password" />
                            </div>
                          <div className="form-check">
                              <label className="form-check-label">
                                  <input type="checkbox" className="form-check-input" />
                                    Remember me
                                </label>
                            </div>
                          <input type="submit" value="Submit" className="btn btn-outline-success size-bt btn-block" />
                        </form>
        <p> Forgot your password?</p>
        <p> Not yet registered? <a href="signup.html"> Let's hook you up with an account. </a> </p>
      </div>
    </div>
  </div>
</div>
<div className="modal fade" id="signupmodal" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
  <div className="modal-dialog" role="document">
    <div className="modal-content">
      <div className="modal-header">
        <h5 className="modal-title" id="exampleModalLabel">Sign in</h5>
        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div className="modal-body">
      <form action="/users/signup" method="POST">
      <div className="form-group">
          <input type="text" className="form-control form-control-lg" placeholder="Username" />
          <i className="glyphicon glyphicon-user form-control-feedback"></i>
      </div>
      <div className="form-group">
          <input type="email" className="form-control form-control-lg" placeholder="Email Address" />
      </div>
      <div className="form-group">
          <input className="password" className="form-control form-control-lg" placeholder="Password" />
      </div>
      <input type="submit" value="Submit" className="btn btn-outline-success size-bt btn-block" />
  </form>
  <p> Have an account? <a href="signin.html"> Sign in </a> </p>
      </div>
    </div>
  </div>
</div>
            </div>

    );
  }
}

export default Navbar;
