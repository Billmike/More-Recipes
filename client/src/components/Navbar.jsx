import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import UserLinks from '../components/navlinks/userLinks';
import GuestLinks from '../components/navlinks/guestLinks';
import { logout } from '../actions/signinRequest';
import logo from '../assets/img/more-recipes-logo.png';

export class Navbar extends Component {
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  logout(event) {
    event.preventDefault();
    this.props.logout();
  }
  render() {
    const { isAuthenticated } = this.props.auth;

    return (
      <nav
        className="navbar
        navbar-expand-lg navbar-light bg-light navbar-fixed-top">
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#startupNavbar"
          aria-controls="navbarTogglerDemo01"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <Link
          to="/"
          className="navbar-brand"
        >
          <img src={logo} alt="More-Recipes" height="80" width="120" />
        </Link>
        <div
          className="collapse navbar-collapse navbar-expand"
          id="startupNavbar"
        >
          {isAuthenticated ? <UserLinks
            logout={this.logout} />
            : <GuestLinks />}
        </div>
      </nav>
    );
  }
}

Navbar.propTypes = {
  auth: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
  return {
    auth: state.auth
  };
};

export default connect(mapStateToProps, { logout })(Navbar);
