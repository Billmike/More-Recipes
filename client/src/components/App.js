import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import Favorite from './Favorite';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import NotFoundPage from './NotFoundPage';

class App extends Component {
    render() {
        return(
            <div>
                <BrowserRouter>
                    <div>
                        <Navbar />
                        <Switch>
                        <Route exact path="/" component={Homepage} />
                        <Route path="/dashboard" component={Dashboard} />
                        <Route path="/favorite" component={Favorite} />
                        <Route path="/add" component={AddRecipe} />
                        <Route path="/edit" component={EditRecipe} />
                        <Route path="/profile" component={ProfilePage} />
                        <Route component={NotFoundPage} />
                    </Switch>
                    </div>
                </BrowserRouter>
            </div>
        )
    }
}

export default App;
