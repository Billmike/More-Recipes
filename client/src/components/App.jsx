import React from 'react';
import { BrowserRouter, browserHistory, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import RecipeDetails from './RecipeDetails';
import Favorite from './Favorites';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import LoginPage from './LoginPage';
import RegistrationPage from './RegistrationPage';
import RequireAuth from '../utils/RequireAuth';
import NotFoundPage from './NotFoundPage';

const App = () => (
  <div>
    <BrowserRouter history={browserHistory}>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/" component={Homepage} />
          <Route path="/dashboard" component={RequireAuth(Dashboard)} />
          <Route path="/favorites" component={RequireAuth(Favorite)} />
          <Route path="/add" component={RequireAuth(AddRecipe)} />
          <Route path="/edit/:id" component={RequireAuth(EditRecipe)} />
          <Route path="/recipe/:id" component={RecipeDetails} />
          <Route path="/profile" component={RequireAuth(ProfilePage)} />
          <Route path="/login" component={LoginPage} />
          <Route path="/register" component={RegistrationPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    </BrowserRouter>
  </div>
);

export default App;
