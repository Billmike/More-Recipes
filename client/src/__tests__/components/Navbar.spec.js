import React from 'react';
import { shallow } from 'enzyme';
import { Navbar } from '../../components/Navbar';
import { auth } from '../fixtures/authUser';

describe('<Navbar />', () => {
  it('Should render the Navbar component correctly', () => {
    const wrapper = shallow(<Navbar auth={auth} />);
    expect(wrapper).toMatchSnapshot();
  });
});
