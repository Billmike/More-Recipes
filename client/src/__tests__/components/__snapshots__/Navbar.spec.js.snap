// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Should render the Navbar component correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light navbar-fixed-top"
>
  <button
    aria-controls="navbarTogglerDemo01"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#startupNavbar"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    <img
      alt="More-Recipes"
      height="80"
      src="test-file-stub"
      width="120"
    />
  </Link>
  <div
    className="collapse navbar-collapse navbar-expand"
    id="startupNavbar"
  >
    <guestLinks />
  </div>
</nav>
`;
